app/controllers/MovementsController.scala

```diff
@@ -38,18 +38,30 @@ class MovementsController @Inject()(
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport {
 
-  private def sort(submissionsWithNotifications: Seq[(SubmissionFrontendModel, Seq[NotificationFrontendModel])]) =
-    submissionsWithNotifications.sortBy(_._1.requestTimestamp)(Ordering[Instant].reverse)
-
   def displayPage(): Action[AnyContent] = authenticate.async { implicit request =>
     val eori = request.user.eori
 
     for {
       submissions <- connector.fetchAllSubmissions(eori)
-      notifications <- Future.sequence(submissions.map(submission => connector.fetchNotifications(submission.conversationId, eori)))
-      submissionsWithNotifications = submissions.zip(notifications.map(_.sorted.reverse))
+      notifications <- connector.fetchAllNotificationsForUser(eori)
+      submissionsWithNotifications = matchNotificationsAgainstSubmissions(submissions, notifications)
+    } yield Ok(movementsPage(sortWithOldestLast(submissionsWithNotifications)))
+  }
 
-    } yield Ok(movementsPage(sort(submissionsWithNotifications)))
+  private def matchNotificationsAgainstSubmissions(
+    submissions: Seq[SubmissionFrontendModel],
+    notifications: Seq[NotificationFrontendModel]
+  ): Seq[(SubmissionFrontendModel, Seq[NotificationFrontendModel])] = {
+    val groupedNotifications: Map[String, Seq[NotificationFrontendModel]] = notifications.groupBy(_.conversationId).withDefaultValue(Seq.empty)
+
+    submissions.map { submission =>
+      (submission, groupedNotifications(submission.conversationId))
+    }
```

Comment:
I think it'll be enough to use `zip` method from `Seq`

https://www.scala-lang.org/api/2.12.3/scala/collection/Seq.html#zip[B](that:scala.collection.GenIterable[B]):Seq[(A,B)]

Reply:
No, it is not. I thought about it, but no. Once notifications are grouped, they're in a Map and they are not ordered to match submissions.
You could `zip` them, but you would have different notifications assigned to submissions every time there is a new notification.

Would be fun though...